quantus/metrics/base.py

```diff
@@ -218,52 +212,14 @@ def __call__(
             custom_batch=custom_batch,
         )
 
-        # Create progress bar if desired.
-        iterator = tqdm(
-            enumerate(
-                zip(
-                    x_batch,
-                    y_batch,
-                    a_batch,
-                    s_batch,
-                    custom_batch,
-                    custom_preprocess_batch,
-                )
-            ),
-            total=len(x_batch),
-            disable=not self.display_progressbar,
-            desc=f"Evaluating {self.__class__.__name__}",
-        )
         self.last_results = [None for _ in x_batch]
-        for id_instance, (
-            x_instance,
-            y_instance,
-            a_instance,
-            s_instance,
-            c_instance,
-            p_instance,
-        ) in iterator:
-            result = self.evaluate_instance(
-                i=int(id_instance),
-                model=model,
-                x=x_instance,
-                y=y_instance,
-                a=a_instance,
-                s=s_instance,
-                c=c_instance,
-                p=p_instance,
-            )
+        iterator = self.get_instance_iterator(data=data)
```

Comment:
We need to write a docsting a la numpydoc for this function

Reply:
done